runtime/compiler/optimizer/EscapeAnalysis.cpp

```diff
@@ -7071,10 +7065,62 @@ void TR_EscapeAnalysis::makeNonContiguousLocalAllocation(Candidate *candidate)
       //printf("Pass: (%d) Non-contiguous allocation found in %s\n", manager()->numPassesCompleted(), comp()->signature());
       }
 
-   // Zero-initialize all the fields
-   //
-   if (candidate->_fields)
+   if (candidate->_node->getOpCodeValue() == TR::newvalue)
+      {
+      // TODO:  Can this code be commoned up with code in lowerNewValue?
+      //
+      TR_OpaqueClassBlock *valueClass = static_cast<TR_OpaqueClassBlock *>(candidate->_node->getFirstChild()->getSymbol()->getStaticSymbol()->getStaticAddress());
+      const TR::TypeLayout* typeLayout = comp()->typeLayout(valueClass);
+
+      // Initialize fields using the values from the newvalue opcode
+      //
+      TR::TreeTop *fieldValueTreeTopCursor = candidate->_treeTop->getPrevTreeTop();
+      TR::TreeTop *fieldStoreTreeTopCursor = candidate->_treeTop;
+
+      for (int i = 1; i < candidate->_node->getNumChildren(); i++)
+         {
+         TR::Node *fieldValueNode = candidate->_node->getChild(i);
+         TR::Node *ttNode = TR::Node::create(TR::treetop, 1);
+         ttNode->setAndIncChild(0, fieldValueNode);
+
+         fieldValueTreeTopCursor = TR::TreeTop::create(comp(), fieldValueTreeTopCursor, ttNode);
+
+         // generate store to the field
+         const TR::TypeLayoutEntry& fieldEntry = typeLayout->entry(i - 1);
+         TR::SymbolReference* symref = comp()->getSymRefTab()->findOrFabricateShadowSymbol(valueClass,
```

Comment:
What is the explanation behind creating this fabricated `symref` ?

Reply:
Are you asking why this `symref` was created but never used?  If so, it looks like that was a mistake on my part in attempting to organize the commits so that the full support for escape analysis of value types accumulated in a reasonably logical fashion.  A later commit in this pull request that adds cold block heapification support for value types actually makes use of the `symref` in a call to [`candidate->findOrSetFieldInfo`](https://github.com/hzongaro/openj9/commit/c2951330f46ea107e6d401e1154d51d3bedfeeef#diff-b5403d98cf7fd820a992fb53dba8c440727fddcd346185442bd1a317c200701cL6781-R6760) and stores it in `field._symRef`.

If you'd like, I can rework these commits so that this `findOrFabricate` call only appears in that later commit.

Or are you looking for additional comments describing the purpose of the call to `findOrFabricateShadowSymbol`?

Reply:
It was the former I think, i.e. I was wondering why it was introduced in this commit. But you explained that as just an issue with organizing the commits.

Reply:
Would you like me to rework those two commits so that the call to `findOrFabricateShadowSymbol` only appears in the commit where it's first needed?